crates/sui-core/src/authority_server.rs

```diff
@@ -596,8 +575,15 @@ impl ValidatorService {
     ) -> Result<tonic::Response<SubmitCertificateResponse>, tonic::Status> {
         let epoch_store = self.state.load_epoch_store_one_call_per_task();
         let certificate = request.into_inner();
-        // The call to digest() assumes the transaction is valid, so we need to verify it first.
-        certificate.validity_check(epoch_store.protocol_config())?;
+        // Being double cautious to also check this for certificates even though we have
+        // already checked when signing transactions.
+        Self::transaction_validity_check(&epoch_store, certificate.data()).tap_err(|err| {
+            // TODO: Send invariant violation metric in release mode; panic in debug mode.
+            error!(
+                "INVARIANT VIOLATION: Transaction validity check failed for certificate: {:?}",
```

Comment:
we haven't verified the cert sig so this is not an invariant violation - a user can submit a cert that fails this